File: library/src/main/java/com/jdroid/java/utils/LocaleUtils.java
package com.jdroid.java.utils;

import java.util.Locale;

public class LocaleUtils {

	public static String getAcceptLanguage() {
		String language = Locale.getDefault().getLanguage();
		String country = Locale.getDefault().getCountry();
		if (StringUtils.isNotBlank(country)) {
			language = language + "-" + country;
		}
		return language;
	}

	public static String getLanguage(String acceptLanguage) {
		return acceptLanguage.split("-")[0];
	}
}

File: jdroid-java-core/src/main/java/com/jdroid/java/utils/LocaleUtils.kt
package com.jdroid.java.utils

import java.util.Locale

object LocaleUtils {

    val acceptLanguage: String
        get() {
            var language = Locale.getDefault().language
            val country = Locale.getDefault().country
            if (StringUtils.isNotBlank(country)) {
                language = "$language-$country"
            }
            return language
        }

    fun getLanguage(acceptLanguage: String): String {
        return acceptLanguage.split("-".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[0]
    }
}
